File: src/main/java/com/aseubel/yusi/repository/UserRepository.java
package com.aseubel.yusi.repository;

import com.aseubel.yusi.pojo.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * @author Aseubel
 * @date 2025/5/7 上午1:14
 */
@Repository
public interface UserRepository extends JpaRepository<User, Long> {

    User findByUserName(String userName);

    User findByUserId(String userId);

    User findByEmail(String email);

    List<User> findByIsMatchEnabledTrue();

    Page<User> findByUserNameContaining(String userName, Pageable pageable);

    Page<User> findByUserIdContaining(String userId, Pageable pageable);
}

File: backend-go/internal/repository/user_repository.go
package repository

import (
	"yusi-go/internal/config"
	"yusi-go/internal/model"
)

type UserRepository struct{}

func NewUserRepository() *UserRepository {
	return &UserRepository{}
}

func (r *UserRepository) Create(user *model.User) error {
	return config.DB.Create(user).Error
}

func (r *UserRepository) FindByUsername(username string) (*model.User, error) {
	var user model.User
	err := config.DB.Where("username = ?", username).First(&user).Error
	return &user, err
}

func (r *UserRepository) FindByUserId(userId string) (*model.User, error) {
	var user model.User
	err := config.DB.Where("user_id = ?", userId).First(&user).Error
	return &user, err
}

func (r *UserRepository) Update(user *model.User) error {
	return config.DB.Save(user).Error
}
